### PR TITLE
[BUGFIX] Persist SQLDatasource splitters on creation

### PR DESCRIPTION
**Pull request by Kilo59**
SQLDatasource asset splitters were not being persisted on creation due to `context._save_project_config()` not being called.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/datasource/fluent/test_postgres_datasource.py [hunk 1]
```diff
@@ -101,6 +107,19 @@ def create_source() -> ContextManager:
 CreateSourceFixture: TypeAlias = Callable[..., ContextManager[PostgresDatasource]]
 
 
+@pytest.fixture
+def mock_test_connection(monkeypatch: pytest.MonkeyPatch):
+    """Monkeypaches the test_connection method of the PostgresDatasource class to return True."""
```
Comment (Kilo59) [inline tests/datasource/fluent/test_postgres_datasource.py]:
```suggestion
    """Patches the test_connection method of the PostgresDatasource class to return True."""
```

File: great_expectations/datasource/fluent/sql_datasource.py [hunk 2]
```diff
@@ -972,7 +977,7 @@ def test_connection(self, test_assets: bool = True) -> None:
     def add_table_asset(  # noqa: PLR0913
         self,
         name: str,
-        table_name: str,
+        table_name: str = "",
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/sql_datasource.py:980]: @cdkini had to adjust these.

File: great_expectations/datasource/fluent/sql_datasource.py [hunk 3]
```diff
@@ -423,6 +424,10 @@ def batch_request_options(self) -> tuple[str, ...]:
     def _add_splitter(self: Self, splitter: Splitter) -> Self:
         self.splitter = splitter
         self.test_splitter_connection()
+        # persist the config changes
+        context: AbstractDataContext | None
+        if context := self._datasource._data_context:
```
Comment (alexsherstinsky) [inline great_expectations/datasource/fluent/sql_datasource.py:429]: 💙

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Review by cdkini — APPROVED** [review]

File: great_expectations/datasource/fluent/sql_datasource.py (hunk 3 above)
Comment (cdkini) [reply great_expectations/datasource/fluent/sql_datasource.py:429]: walrus!

**Pull request merged**